shared/src/commonMain/kotlin/io/customer/shared/internal/InternalCustomerIOApi.kt

```diff
@@ -0,0 +1,11 @@
+package io.customer.shared.internal
+
+/**
+ * Marks declarations that should be used by only Customer.io classes. Client apps should avoid
+ * using them as they may change without any announcement.
+ */
+@RequiresOptIn(
+    message = "This is internal API for CustomerIO. Do not depend on this API in your own client code.",
+    level = RequiresOptIn.Level.ERROR
+)
+annotation class InternalCustomerIOApi
```

Comment:
Since making our APIs `internal` will prevent our SDKs from using them, we'll still have to expose some classes/methods that we would not want our customers to use. This annotation might help us warning about them.

But we couldn't test this on iOS yet, will do that soon. If it works well, all good, if not, we'll remove it from here and move to Android SDK only.

Reply:
Although I like the idea, I do think by using this annotation we are adding more work to our team in the future. 

We need to remember to include this annotation for every class/method/property that is `public` scope, but not part of our official SDK's API. That seems error-prone in that we will probably forget to include the annotation on something in the future. 

I think we could come up with a solution that's more simple and requires less maintenance.